src/main/asciidoc/reference/projections.adoc

```diff
@@ -0,0 +1,118 @@
+[[reference_programming-projections]]
+=== Projections
+
+Spring Data Repositories usually return the domain model when using query methods. However, sometimes, you may need to alter the view of that model for various reasons. In this section, you will learn how to define projections to serve up simplified and reduced views of resources.
+
+Look at the following domain model:
+
+[source,java]
+----
+@NodeEntity
+public class Cinema {
+
+  private Long id;
+  private String name, location;
+
+  @Relationship(type = "VISITED", direction = Relationship.INCOMING)
+  private Set<User> visited = new HashSet<>();
+
+  @Relationship(type = "BLOCKBUSTER", direction = Relationship.OUTGOING)
+  private Movie blockbusterOfTheWeek;
+  …
+}
+----
+
+This `Cinema` has several attributes:
+
+* `id` is the graph id
+* `name` and `location` are data attributes
+* `visited` and `blockbusterOfTheWeek` are links to other domain objects
+
+Now assume we create a corresponding repository as follows:
+
+[source,java]
+----
+public interface CinemaRepository extends Neo4jRepository<Cinema, Long> {
+
+  Cinema findByName(String name);
+}
+----
+
+Spring Data will return the domain object including all of its attributes, including all the users that visited this cinema. That can be a big amount of data and can lead to performance issues.
+
+There are several ways to avoid that :
+
+* use a custom `depth` for loading (see <<reference_derived-queries>>)
+* use a custom annotated query (see <<reference_programming_model_annotatedQueries>>)
+* use a projection
+
+.Simple Projection
+====
+[source,java]
+----
+public interface CinemaNameAndBlockbuster {  <1>
+
+	public String getName(); <2>
+	public Movie getBlockbusterOfTheWeek();
+}
+----
+This projection has the following details:
+
+<1> A plain Java interface making it declarative.
+<2> Only some attributes of the entity are exported.
+====
+
+The `CinemaNameAndBlockbuster` projection only has getters for `name` and `blockbusterOfTheWeek` meaning that it will not serve up any user information. The query method definition returns in this case `CinemaNameAndBlockbuster` instead of `Cinema`.
+
+[source,java]
+----
+interface CinemaRepository extends Neo4jRepository<Cinema, Long> {
+
+  CinemaNameAndBlockbuster findByName(String name);
+}
+----
+
+Projections declare a contract between the underlying type and the method signatures related to the exposed properties. Hence it is required to name getter methods according to the property name of the underlying type. If the underlying property is named `name`, then the getter method must be named `getName` otherwise Spring Data is not able to look up the source property. This type of projection is also called _closed projection_. Closed projections expose a subset of properties hence they can be used to optimize the query in a way to reduce the selected fields from the data store. The other type is, as you might imagine, an _open projection_.
```

Comment:
Should there be a note that closed projections are not yet supported in SDN? Or is this doc one-to-one copy from other SD docs?

Reply:
Closed projections are supported. 
What is not is the query optimization. The way things are done currently rely on the OGM session, which acts as a cache. Triggering optimized requests with partial entity information requires some more work : 
* enrich the OGM cypher request builders, or add one in SDN
* see how it affects the session caching. not sure that having partially hydrated entities in the cache is a good thing

I'll change this section of the documentation to make it clearer to users and explain further the custom query alternative.